src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/AdaptiveChainPruner.java

```diff
@@ -29,41 +33,105 @@ public AdaptiveChainPruner(final double initialErrorProbability, final double lo
 
-        return likelyErrorChains(chains, graph, errorRate);
+        return likelyErrorChains(chains, graph, errorRate).stream().filter(c -> !c.getEdges().stream().anyMatch(BaseEdge::isRef)).collect(Collectors.toList());
     }
 
     private Collection<Path<V,E>> likelyErrorChains(final List<Path<V, E>> chains, final BaseGraph<V,E> graph, final double errorRate) {
-        final Map<Path<V,E>, Double> chainLogOdds = chains.stream()
+        /////// NEW
+
+        // pre-compute the left and right log odds of each chain
+        final Map<Path<V,E>, Pair<Double, Double>> chainLogOdds = chains.stream()
                 .collect(Collectors.toMap(c -> c, c-> chainLogOdds(c, graph, errorRate)));
 
-        final Set<Path<V,E>> result = new HashSet<>(chains.size());
+        // compute correspondence of vertices to incident chains with log odds above the seeding and extending thresholds
+        final Multimap<V, Path<V,E>> vertexToSeedableChains = ArrayListMultimap.create();
+        final Multimap<V, Path<V,E>> vertexToGoodIncomingChains = ArrayListMultimap.create();
+        final Multimap<V, Path<V,E>> vertexToGoodOutgoingChains = ArrayListMultimap.create();
+
+        for (final Path<V,E> chain : chains) {
+            if (chainLogOdds.get(chain).getRight() >= logOddsThreshold) {
+                vertexToGoodIncomingChains.put(chain.getLastVertex(), chain);
+            }
+
+            if (chainLogOdds.get(chain).getLeft() >= logOddsThreshold) {
+                vertexToGoodOutgoingChains.put(chain.getFirstVertex(), chain);
+            }
+
+            // seed-worthy chains must pass the more stringent seeding log odds threshold on both sides
+            // in addition to that, we only seed from vertices with multiple such chains incoming or outgoing (see below)
+            if (chainLogOdds.get(chain).getRight() >= seedingLogOddsThreshold && chainLogOdds.get(chain).getLeft() >= seedingLogOddsThreshold) {
+                vertexToSeedableChains.put(chain.getFirstVertex(), chain);
+                vertexToSeedableChains.put(chain.getLastVertex(), chain);
+            }
+        }
+
+        // find a subset of good vertices from which to grow the subgraph of good chains.
+        final Queue<V> verticesToProcess = new ArrayDeque<>();
+        final Path<V,E> maxWeightChain = getMaxWeightChain(chains);
+        verticesToProcess.add(maxWeightChain.getFirstVertex());
+        verticesToProcess.add(maxWeightChain.getLastVertex());
+
+        // look for vertices with two incoming or two outgoing chains (plus one outgoing or incoming for a total of 3 or more) with good log odds to seed the subgraph of good vertices
+        // the logic here is that a high-multiplicity error chain A that branches into a second error chain B and a continuation-of-the-original-error chain A'
+        // may have a high log odds for A'.  However, only in the case of true variation will  multiple branches leaving the same vertex have good log odds.
```

Comment:
Interesting, I still think this can be fooled if there is a bubble of low weight in an error branch for which two low weight edges of equal weight split and merge and ultimately are connected to the reference only by bogus chains.

Reply:
This is also averted by the stricter seeding log odds threshold.  I wrote a unit test for it.